src/libs/ReportUtils.ts

```diff
@@ -3430,14 +3445,66 @@ function shouldHideReport(report: OnyxEntry<Report>, currentReportId: string): b
     return parentReport?.reportID !== report?.reportID && !isChildReportHasComment;
 }
 
+/**
+ * Checks to see if a report's parentAction is a money request that contains a violation
+ */
+function doesTransactionThreadHaveViolations({
+    report,
+    transactionViolations,
+    parentReportAction,
+}: {
+    report: Report;
+    transactionViolations?: OnyxCollection<TransactionViolation[]>;
+    parentReportAction: ReportAction;
+}): boolean {
+    if (parentReportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.IOU) {
+        return false;
+    }
+    const {IOUTransactionID, IOUReportID} = parentReportAction.originalMessage ?? {};
+    if (!IOUTransactionID || !IOUReportID) {
+        return false;
+    }
+    if (!isCurrentUserSubmitter(IOUReportID)) {
+        return false;
+    }
+    if (report.stateNum !== CONST.REPORT.STATE_NUM.OPEN && report.stateNum !== CONST.REPORT.STATE_NUM.PROCESSING) {
+        return false;
+    }
+    return hasViolation(IOUTransactionID, transactionViolations);
+}
+
+/**
+ * Checks to see if a report contains a violation
+ */
+function hasViolations(reportID: string, transactionViolations: OnyxCollection<TransactionViolation[]>): boolean {
+    const transactions = TransactionUtils.getAllReportTransactions(reportID);
+    return transactions.some((transaction) => hasViolation(transaction.transactionID, transactionViolations));
+}
+
 /**
  * Takes several pieces of data from Onyx and evaluates if a report should be shown in the option list (either when searching
  * for reports or the reports shown in the LHN).
  *
  * This logic is very specific and the order of the logic is very important. It should fail quickly in most cases and also
  * filter out the majority of reports before filtering out very specific minority of reports.
  */
-function shouldReportBeInOptionList(report: OnyxEntry<Report>, currentReportId: string, isInGSDMode: boolean, betas: Beta[], policies: OnyxCollection<Policy>, excludeEmptyChats = false) {
+function shouldReportBeInOptionList({
+    report,
+    currentReportId,
+    isInGSDMode = false,
+    betas,
+    policies,
+    excludeEmptyChats = false,
+    doesReportTransactionThreadHaveViolations,
```

Comment:
Let's simplify here by removing "transaction thread" which isn't relevant here
```suggestion
    hasViolations,
```

Reply:
hasViolations is already defined in the outer scope, so this causes a linter error.

Reply:
Even if it's passed as an individual param? If so, I think it's worth suppressing the error, or calling this `doesReportHaveViolations`

Reply:
done